R/grade_result.R

```diff
@@ -47,18 +77,6 @@ grade_result <- function(
       break
     }
   }
-
-  message <- glue_message(
-    {if (final_result$correct) glue_correct else glue_incorrect}, # nolint
-    .is_match = found_match,
-    .is_correct = final_result$correct,
-    .message = final_result$message,
-    .correct = correct,
-    .incorrect = incorrect
-  )
-
-  return(graded(
-    correct = final_result$correct,
-    message = message
-  ))
+  
+  list(matching_conditional = found_match, result = final_result)
```

Comment:
find_first_condition

return instead of a list, just final_result if found, or NULL otherwise.

Other functions update to handle that signature.